CMSgov/onemac
# Oy2 20375

Story: https://qmacbis.atlassian.net/browse/OY2-20375
Endpoint: See github-actions bot comment

### Details

Require new temporary extensions from New Submission button in package view to enter the approved initial or renewal waiver.

### Changes

- added parent validation lambda for temp waiver
- added config to temp extension types to use existing parent id entry infrastructure
- updated componentId to allow passing in an id so that parent component id can differ from the actual component id on the form when using more than one `ComponentId` component on a page

### Test Plan

1. Login as state submitter
2. Navigate to package view and New Submission -> Waiver Action -> Request Temporary Extension
3. Verify title and subtext of parent id field per ACs
4. Enter any parent id that is not an approved initial or renewal waiver
5. Verify the error message matches the AC
